test/qpu_interface.jl

```diff
@@ -269,59 +270,46 @@ end
-        access_token = expected_access_token,
-        requestor = requestor,
-    )
-    status, histogram = get_status(test_client, "jobID not used in this test")
-    @test get_status_type(status) == Snowflurry.failed_status
-    @test get_status_message(status) == "mocked"
+    statuses = [
+        (
+            "jobID not used in this test",
+            stubFailedStatusResponse(),
+            Snowflurry.failed_status,
+            "mocked",
+        ),
+        ("jobID_queued", stubQueuedStatusResponse(), Snowflurry.queued_status, ""),
+        ("jobID_running", stubRunningStatusResponse(), Snowflurry.running_status, ""),
+        ("jobID_cancelled", stubCancelledResultResponse(), Snowflurry.cancelled_status, ""),
+        ("jobID_successful", stubSuccessStatusResponse(), Snowflurry.succeeded_status, ""),
+        ("jobID_invalid", stubFailedStatusResponse(), Snowflurry.failed_status, "invalid"),
+    ]
 
-    test_get = stub_response_sequence([
-        # Simulate a response containing an invalid job status.
-        stubStatusResponse("not a valid status"),
-    ])
+    for (job_id, stub_response, expected_status_type, expected_message) in statuses
+        requestor =
+            MockRequestor(make_request_checker(stub_response, expected_realm), test_post)
 
-    requestor = HTTPRequestor(test_get, test_post)
-    test_client = Client(
-        host = expected_host,
-        user = expected_user,
-        access_token = expected_access_token,
-        requestor = requestor,
-    )
-    @test_throws ArgumentError get_status(test_client, "jobID not used in this test")
-
-    malformedResponse = stubFailedStatusResponse()
-    # A failure response _should_ have a 'message' field but, if things go very
-    # wrong, the user should still get something useful.
-    body = "{\"status\":{\"type\":\"FAILED\"},\"these aren't the droids you're looking for\":\"*waves-hand*\"}"
-    malformedResponse.body = collect(UInt8, body)
-    test_get = stub_response_sequence([malformedResponse])
-    requestor = HTTPRequestor(test_get, test_post)
-    test_client = Client(
-        host = expected_host,
-        user = expected_user,
-        access_token = expected_access_token,
-        requestor = requestor,
-    )
-    status, histogram = get_status(test_client, "jobID not used in this test")
-    @test status.type == Snowflurry.failed_status
-    @test status.message != ""
-end
+        test_client = Client(
+            host = expected_host,
+            user = expected_user,
+            access_token = expected_access_token,
+            realm = expected_realm,
+            requestor = requestor,
+        )
+
+        status, histogram = Snowflurry.get_status(test_client, job_id)
 
+        @test get_status_type(status) == expected_status_type
+
+        if expected_message == "invalid"
+            @test status.message != ""
+        elseif expected_message != ""
+            @test get_status_message(status) == expected_message
+        end
```

Comment:
no need for branching, the expected_message is always what get_status_message should be

```suggestion
       @test get_status_message(status) == expected_message
```